### PR TITLE
ThemeGen tool to generated the control and page arrays for ThmUtil

### PR DESCRIPTION
**Pull request by barnson**

### CODE REVIEW

File: src/tools/thmgen/Templates/Controls.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿// DO NOT MODIFY THIS FILE. IT IS GENERATED BY A TOOL.
```
Comment (robmen) [inline src/tools/thmgen/Templates/Controls.cpp:1]: Check for the "generated" tag that Visual Studio recognizes.

File: src/tools/thmgen/thmgen.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+            return Messaging.Instance.LastErrorNumber;
+        }
+
+        private void GenerateThemeHelpers()
+        {
+            var outputHeaderFile = Path.ChangeExtension(this.commandLine.OutputFile ?? "theme.h", ".h");
+            var outputModuleFile = Path.ChangeExtension(outputHeaderFile, ".cpp");
+            var prefix = this.commandLine.Prefix ?? String.Empty;
+            var pch = this.commandLine.PchName ?? "stdafx.h";
```
Comment (robmen) [inline src/tools/thmgen/thmgen.cs:104]: Might consider just making this the "header to get dutil stuff included". Precompiled header is the most common case...

File: src/tools/thmgen/thmgen.csproj [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      <Generator>ResXFileCodeGenerator</Generator>
+      <LastGenOutput>ThmGenStrings.Designer.cs</LastGenOutput>
+    </EmbeddedResource>
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="System" />
+    <Reference Include="System.XML" />
+    <Reference Include="System.Xml.Linq" />
+    <Reference Include="wix">
+      <HintPath>..\..\..\..\..\dev\wix4\wix.dll</HintPath>
```
Comment (robmen) [inline src/tools/thmgen/thmgen.csproj:51]: Probably not the right path.

File: src/tools/thmgen/thmgen.exe.manifest [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!--
+  <copyright file="smoke.exe.manifest" company="Outercurve Foundation">
```
Comment (robmen) [inline src/tools/thmgen/thmgen.exe.manifest:3]: `thmgen` not `smoke`, eh?

File: src/tools/thmgen/thmgen.csproj [hunk 5]
```diff
@@ -0,0 +1,64 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+  <copyright file="smoke.csproj" company="Outercurve Foundation">
```
Comment (robmen) [inline src/tools/thmgen/thmgen.csproj:3]: Not `smoke`.

File: src/tools/thmgen/thmgen.cs [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                return String.Format("L\"{0}\"", name);
+            });
+        }
+
+        private static IEnumerable<string> GetControlsInitialization(IEnumerable<XElement> controls, string prefix)
+        {
+            return controls.Select(control =>
+            {
+                var name = control.Attribute("Name").Value;
+                return String.Format("{{ {0}Controls::{1}, L\"{1}\" }}", prefix, name);
```
Comment (barnson) [inline src/tools/thmgen/thmgen.cs:161]: Investigate whether we need to block out ids for things like IDCANCEL that Windows will send.

**Comment by barnson** [comment]
GitHub doesn't like branch names that differ in case...

**Pull request closed without merging**